### PR TITLE
feat(docker): refactor for docker registries

### PR DESCRIPTION
**Pull request by giokara-work**
## Changes:

Small refactor as requested in https://github.com/renovatebot/renovate/pull/9813.

## Context:

Refactor as requested in https://github.com/renovatebot/renovate/pull/9813.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
Can you remove the quay.io relates changes from this pr, so it is only a refactor pr. after this you can update the original pr to only include the quay.io related changes.

**Comment by viceice** [comment]
Sorry if i wasn't clear enough, but we like to have the required docker datasource refactoring in one pr and then the quay.io additions in an additional pr.

**Comment by giokara-oqton** [comment]
Sorry - I misunderstood. Removed all quay related changes.

**Review by viceice — COMMENTED** [review]
- Move `id` and `http` to `./common.ts` too.
- Move all interfaces / types to `./types.ts`.

File: lib/datasource/docker/index.ts [hunk 1]
```diff
@@ -52,15 +48,6 @@ export const defaultConfig = {
...
-const http = new Http(id);
-
-const ecrRegex = /\d+\.dkr\.ecr\.([-a-z0-9]+)\.amazonaws\.com/;
-
-export interface RegistryRepository {
-  registry: string;
-  repository: string;
-}
-
 export function getRegistryRepository(
```
Comment (viceice) [inline lib/datasource/docker/index.ts:51]: move this and all unecported function to `./common.ts`

**Comment by giokara-oqton** [comment]
Anything else that I can do to help finish this?

**Comment by viceice** [comment]
> Anything else that I can do to help finish this?

Next time use the (re)request review function to catch our attention
image

Otherwise we don't know that you've fixed the requested issues.

File: lib/datasource/docker/common.ts [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+import { ensureTrailingSlash, trimTrailingSlash } from '../../util/url';
+import { MediaType, RegistryRepository } from './types';
+
+export const id = 'docker';
+export const http = new Http(id);
+
+export const ecrRegex = /\d+\.dkr\.ecr\.([-a-z0-9]+)\.amazonaws\.com/;
+export const defaultRegistryUrls = ['https://index.docker.io'];
+
+export async function getECRAuthToken(
```
Comment (viceice) [inline lib/datasource/docker/common.ts:21]:
```suggestion
async function getECRAuthToken(
```

**Review by viceice — APPROVED** [review]

**Review by viceice — CHANGES_REQUESTED** [review]
Please rename the `index.spec.ts` to `common.spec.ts` and add a new `index.spec.ts to test the remaining functions in `index.ts`

**Comment by giokara-oqton** [comment]
Not really sure why the codecov tests for https://github.com/renovatebot/renovate/pull/9891/commits/691f4e34b33e6286779e596775965ebb0b7ad0da failed above. I can't seem to access the codecov page either, any pointers on what is missing a test?

**Review by viceice — APPROVED** [review]

**Pull request merged**